Civi/Api4/Action/CustomGroup/Get.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+namespace Civi\Api4\Action\CustomGroup;
+
+use Civi\Api4\Generic\Result;
+use Civi\Api4\Generic\Traits\ArrayQueryActionTrait;
+
+/**
+ * @inheritDoc
+ *
+ */
+class Get extends \Civi\Api4\Generic\DAOGetAction {
+  use ArrayQueryActionTrait;
+
+  /**
+   * @var bool
+   *
+   * Should we use the in-memory cache to answer
+   * this request?
+   *
+   * If unset, will be determined automatically based
+   * on the complexity of the request
+   */
+  protected ?bool $useCache = NULL;
+
+  /**
+   * @param \Civi\Api4\Generic\Result $result
+   *
+   * Most of the time uses the standard DAOGetAction implementation
+   *
+   * However - for simple queries we can use the in-memory cache to
+   * avoid hitting the database
+   */
+  protected function getObjects(Result $result) {
+    if (is_null($this->useCache)) {
+      $this->useCache = !$this->needDb();
+    }
+    if ($this->useCache) {
+      $this->getFromCache($result);
+      return;
+    }
+    parent::getObjects($result);
+  }
+
+  protected function needDb() {
```

Comment:
> Clauses that use field-to-field comparisons

Do you mean they don't technically "need" the DB? That makes sense but at this stage I would be happy to leave those out of scope because it like a chunk more work for minimal gain?